tiledb/ml/readers/_tensor_schema.py

```diff
@@ -31,60 +32,28 @@ class TensorSchema(ABC):
     A class to encapsulate the information needed for mapping a TileDB array to tensors.
     """
 
-    def __init__(
-        self,
-        array: tiledb.Array,
-        key_dim: Union[int, str] = 0,
-        fields: Sequence[str] = (),
-        transform: Optional[Callable[[Tensor], Tensor]] = None,
-    ):
-        """Create a TensorSchema.
-
-        :param array: TileDB array to read from.
-        :param key_dim: Name or index of the key dimension. Defaults to the first dimension.
-        :param fields: Attribute and/or dimension names of the array to read. Defaults to
-            all attributes.
-        :param transform: Function to transform tensors.
-        """
-        all_attrs = [array.attr(i).name for i in range(array.nattr)]
-        all_dims = [array.dim(i).name for i in range(array.ndim)]
-        dims = []
-        if fields:
-            attrs = []
-            for field in fields:
-                if field in all_attrs:
-                    attrs.append(field)
-                elif field in all_dims:
-                    dims.append(field)
-                else:
-                    raise ValueError(f"Unknown attribute or dimension '{field}'")
-        else:
-            fields = attrs = all_attrs
-
-        ned = list(array.nonempty_domain())
-        key_dim_index = key_dim if isinstance(key_dim, int) else all_dims.index(key_dim)
-        if key_dim_index > 0:
-            # Swap key dimension to first position
-            all_dims[0], all_dims[key_dim_index] = all_dims[key_dim_index], all_dims[0]
-            ned[0], ned[key_dim_index] = ned[key_dim_index], ned[0]
-
-        self._array = array
-        self._key_dim_index = key_dim_index
-        self._ned = tuple(ned)
-        self._all_dims = tuple(all_dims)
-        self._fields = tuple(fields)
-        self._query_kwargs = {"attrs": tuple(attrs), "dims": tuple(dims)}
-        self._transform = transform
+    def __init__(self, array_params: ArrayParams):
+        self._array = array_params._tensor_schema_kwargs["array"]
+        self._fields = array_params._tensor_schema_kwargs["fields"]
+        self._key_dim_index = array_params._tensor_schema_kwargs["key_dim_index"]
+        self._ned = array_params._tensor_schema_kwargs["ned"]
+        self._all_dims = array_params._tensor_schema_kwargs["all_dims"]
+        self._query_kwargs = array_params._tensor_schema_kwargs["query_kwargs"]
 
     @property
     def fields(self) -> Sequence[str]:
         """Names of attributes and dimensions to read."""
-        return self._fields
+        return cast(Sequence[str], self._fields)
 
     @property
     def field_dtypes(self) -> Sequence[np.dtype]:
         """Dtypes of attributes and dimensions to read."""
-        return tuple(map(self._array.schema.attr_or_dim_dtype, self._fields))
+        return tuple(
+            map(
+                self._array.schema.attr_or_dim_dtype,
+                self._fields,
+            )
+        )
```

Comment:
[formatting] this can be one line as it was after removing the trailing comma after `self._fields`